Tools/AP_Periph/AP_Periph.h

```diff
@@ -58,6 +60,16 @@ class AP_Periph_FW {
     AP_Baro baro;
 #endif
 
+#ifdef HAL_PERIPH_ENABLE_BATTERY
+    void handle_battery_failsafe(const char* type_str, const int8_t action) { }
+
+    AP_BattMonitor battery{0, FUNCTOR_BIND_MEMBER(&AP_Periph_FW::handle_battery_failsafe, void, const char*, const int8_t), nullptr};
+
+    uint32_t battery_last_read_ms;
```

Comment:
I'd prefer to put the battery vars in a battery struct, like we did for msp and adsb. Not a big thing, just helps keep things tidy

Reply:
done, moved into a struct